Added ability to pull date and location from EXIF on images.

gonzonia opened the pull request:
I've added ability to pull date and location from EXIF on images. I just realized there are a couple of more edits to make to the Readme, but all the config info is there. I'm submitting this as a pull request because your repository is more likely to be found and used.

gonzonia committed:
Adding ability to display date/location from Exif data

Adding ability to display date/location from Exif data

gonzonia committed:
Update README.md

gonzonia committed:
Update README.md

gonzonia committed:
Update README.md

gonzonia committed:
Update MMM-ImagesPhotos.js

Added checks for API Key missing or not working.

gonzonia committed:
Merge branch 'master' of https://github.com/gonzonia/MMM-ImagesPhotos

gonzonia committed:
Update MMM-ImagesPhotos.js

Put API checks in non-FS section

sdetweil commented:
two things

1. the exifr lib is not listed in package.json , need npm install exifr, which updates package.json
2. in MMM-ImagePhotos.js there are two lines with L.log   should be Log.log

the data returned from node helper doesn't contain the date of the image (copied .png from downloads folder) 
I don't know if there IS some exif data..

gonzonia committed:
Updating package.json to include exifr, fixing typos, adding catch for no date

Added exifr to the package.json
Fixed L.log typo to Log.log
Added additional check in node_helper and MMM-ImagesPhotos for no date found. Node_helper will return a blank and MMM-ImagesPhotos will check for blank as well as undefined and null

gonzonia commented:
Good catch. I adjusted accordingly. I've added some additional checks in case it doesn't find an exif date.

gonzonia committed:
Update node_helper.js

Adde HEIC change from master

gonzonia committed:
Update README.md

gonzonia committed:
Update MMM-ImagesPhotos.js

Adding error handling that was in upstream master

sdetweil commented:
there are version conflicts, can you fix those.

gonzonia committed:
Update .gitignore

Added package-lock.json

gonzonia committed:
Merge branch 'master' into master

gonzonia commented:
I think that did it. I was just adding the last few commits you made into my code.

sdetweil commented:
readme url

is there any readme update about the exif info?

gonzonia commented:
Yes, in my repository I've updated the readme (it is also updated to reference installing from my repository since it was unique to my changes).

sdetweil commented:
urs, but we would be merging into my repo

gonzonia committed:
Update README.md

Chaged back to sdetweil's information for installing. 

Note: I'm getting an error on load after latest MM update. 

 Error when loading MMM-ImagesPhotos: Unexpected identifier 'config'

gonzonia commented:
I've just updated the readme to reflect your repo info again. 
For some reason I'm getting an error on load though after the latest MM update -`Error when loading MMM-ImagesPhotos: Unexpected identifier 'config' `

sdetweil commented:
can you show the module config?

gonzonia commented:
Sure. Thanks for taking a look. I figured that's where it was too, but when I switch to your fork it doesn't kick up an error with the same config. 
``` 
`{
		module: "MMM-ImagesPhotos",
		position: "fullscreen_below",
		classes: 'SceneFamily SceneSam',
		config: {
			opacity: 0.9,
			animationSpeed: 0,
			updateInterval: 120000, //How often to 
			getInterval: 6000000, //How often to check directory?
			maxWidth: "100%",
			maxHeight: "100%",
			retryDelay: 2500,
			fill: true,   // fill around image with blurred copy of image
			blur: 10,   // only used if fill is true
			sequential: false,   // process the image list randomly
			sleepList: "SCREENSAVE_ENABLED", //Suspend when these notifications are recieved
			wakeList: "SCREENSAVE_DISABLED", //Resume when these notifications are recieved
			showExifDate: true,
			showDateLabel: false,
			BigDataGeoAPI:"XXXXXXXXXXXXX",
			showCity: true,
		}
	},`
```

sdetweil commented:
npm run config:check

reports it too?

we updated the syntax checker to catch literals in the wrong place, but i havent seen this on my system

gonzonia commented:
No. 
pi@MagicMirror:~/MagicMirror $ npm run config:check

> magicmirror@2.32.0 config:check
> node js/check_config.js

[2025-07-01 17:56:29.538] [INFO]  Checking config file /home/pi/MagicMirror/config/config.js ... 
[2025-07-01 17:56:29.791] [INFO]  Your configuration file doesn't contain syntax errors :) 
[2025-07-01 17:56:29.793] [INFO]  Checking modules structure configuration ... 
[2025-07-01 17:56:30.034] [INFO]  Your modules structure configuration doesn't contain errors :)

gonzonia commented:
I found the problem. At some point I messed up something in node_helper with a bad cut and paste. Working on fixing it now.

gonzonia committed:
Fixing bad cut and paste that duplicated code.

gonzonia committed:
Fixing the rest of the bad cut and paste

Fixing the rest of the bad cut and paste

gonzonia committed:
Added check for Exif output
